### ISSUE TITLE
Stream support

### ISSUE THREAD

**Issue by Sander-Toonen** [1up-lab/OneupFlysystemBundle#56]
Is there a way to hook a filesystem into for instance the liipImagineBundle configuration? With Gaufrette a stream wrapper can be configured for that purpose. I found that there is a project that enables stream wrappers for Flysystem filesystems (https://github.com/twistor/flysystem-stream-wrapper/).

**Comment by sheeep** [comment]
Hi @Sander-Toonen 

I like the idea to include the possibility for `stream wrapper` support and therefore started to work on this.
You can find it in the [`feature-stream_wrapper` branch](https://github.com/1up-lab/OneupFlysystemBundle/tree/feature-stream_wrapper).

It is possible to configure a stream wrapper per filesystem config entry. It is however not yet registered.
The main problem I faced while implementing was finding the place to call `FlysystemStreamWrapper::register`. I think we can not add it to the extension, as it will not be executed, once the container is compiled. IIRC, the only solution would be to put it in a `kernel.init` event. Any idea?

**Comment by Sander-Toonen** [comment]
Is there such a thing as the ```kernel.init``` event? You could hook into ```kernel.request``` but that wouldn't cover console commands and custom scripts.

Another option would be to implement the ```boot()``` function of the bundle class. The bundle class is container aware thus it should be able to use the configuration from there.

**Comment by sheeep** [comment]
Whoops, you're right. There is [no such thing](http://symfony.com/doc/current/components/http_kernel/introduction.html#component-http-kernel-event-table) as a `kernel.init` event apparently. Isn't the `boot` method already executed, when the container is compiled?

**Comment by Sander-Toonen** [comment]
The boot method of the bundle is executed every time the boot method of the kernel is called (whether the container gets compiled or not). If the container needs to be compiled, this is done before the execution of the bundles boot methods.

**Comment by foaly-nr1** [comment]
+1 for this feature

**Comment by apsylone** [comment]
+1

**Comment by Sander-Toonen** [comment]
@bytehead There is a pull request and it looks good. Anything we can do to get this feature merged?

**Comment by bytehead** [comment]
Hi @Sander-Toonen, did you already test the PR?

**Comment by Sander-Toonen** [comment]
@bytehead Yes, I've tested it by replacing KnpGaufretteBundle by OneupFlysystemBundle + this PR in a project that uses stream wrappers in several places and in multiple ways. I only found one edge case problem that causes stream wrappers to be registered twice when executing the cache:clear command. This however probably needs to be resolved in ``twistor/flysystem-stream-wrapper``. I'll report that issue there.

I have confidence enough in this PR to run it in production but a look from someone with some more knowledge on this bundle certainly wouldn't hurt.
